pyne/mesh.py

```diff
@@ -1231,24 +1231,35 @@ def tag_cell_fracs(self, cell_fracs):
         # creat tag frame with default value
         cell_largest_frac_number = [-1] * num_vol_elements
         cell_largest_frac = [0.0] * num_vol_elements
-        voxel_cell_number = np.empty(shape=(num_vol_elements,max_num_cells),
-                                     dtype=int)
-        voxel_cell_fracs = np.empty(shape=(num_vol_elements,max_num_cells),
-                                    dtype=float)
+        if max_num_cells == 1:
```

Comment:
I'm not totally clear on why thy array needs to be one dimensional if the number of cells is 1, but I'm wondering if [this post](https://stackoverflow.com/questions/42882842/whats-the-difference-between-n-and-n-1-in-numpy) might help.

Mostly I'm thinking these can be treated as two dimensional arrays until the very end of the function when the mesh.tag function is called. Right before this they can be reshaped. This would reduce the lines of code above that considerably and keep the indexing the same for most of the function. 

So something like: 

```
# no changes in code up to here

# line 1263
#create the tags
if max_num_cells == 1: 
    voxel_cell_numer.reshape((num_vol_elements,))
    voxel_cell_fracs.reshape((num_vol_elements,))

# previous tag calls as they were before
  self.tag(name='cell_number', value=voxel_cell_number,
                   doc='cell numbers of the voxel, -1 used to fill vacancy',
                  tagtype=IMeshTag, size=max_num_cells, dtype=int)
self.tag(name='cell_fracs', value=voxel_cell_fracs,
                  tagtype=IMeshTag, doc='volume fractions of each cell in the '
                                        'voxel, 0.0 used to fill vacancy',
                  size=max_num_cells, dtype=float)

```
(Sorry, indentation isn't great. In a hurry)